### PR TITLE
Implemented the improvements, removed boost:foreach and so on

### PR DESCRIPTION
**Pull request by pnikiel**
Most of the ideas come from Ben from a past review, and some come from me.

### CODE REVIEW

File: include/nodemanagerbase.h [hunk 1]
```diff
@@ -33,7 +33,10 @@ class NodeManagerConfig
 class NodeManagerBase: public NodeManagerConfig
 {
 public:
-    NodeManagerBase( const char* uri, bool sth, int hashtablesize );
+    NodeManagerBase(
+            const char* sNamespaceUri,
+            bool        firesEvents = OpcUa_False,
+            OpcUa_Int32 nHashTableSize = 9973 );
```
Comment (ben-farnham) [inline include/nodemanagerbase.h:39]: what's the magic number? 
Maybe name it with a
`const static OpcUa_Int32 g_sFaxNumberOfTheBeast = 9973`
and so the param list could be a more readable
`OpcUa_Int32 nHashTableSize = g_sFaxNumberOfTheBeast`

File: src/nodemanagerbase.cpp [hunk 2]
```diff
@@ -19,54 +19,50 @@
...
-    {
-        if (node->nodeId() == nodeId)
-            return node;
-    }
-    return 0; // not found
+    auto iterator = std::find_if(
+            m_listNodes.begin(),
+            m_listNodes.end(),
+            [nodeId](UaNode* node){return node->nodeId() == nodeId;});
+    if (iterator != m_listNodes.end())
```
Comment (ben-farnham) [inline src/nodemanagerbase.cpp]: It's maybe a bit C++ golf-ish but I often use
`return (iterator != m_listNodes.end()) ? *iterator : nullptr`
just as an alternative

File: src/nodemanagerbase.cpp [hunk 3]
```diff
@@ -19,54 +19,50 @@
...
+    m_nameSpaceUri(sNamespaceUri)
 {
 
 }
 
 NodeManagerBase::~NodeManagerBase()
 {
-
-
-    std::cout << __FUNCTION__ << std::endl;
```
Comment (ben-farnham) [inline src/nodemanagerbase.cpp:43]: maybe worth a 
`LOG(TRC) << __FUNCTION__`
in new code

**Review by ben-farnham — COMMENTED** [review]
I like it - go C++11.

File: include/nodemanagerbase.h (hunk 1 above)
Comment (pnikiel) [reply include/nodemanagerbase.h:39]: @ben-farnham "hash table size", to my mind, is a common term in data structures, so I'm not particularly willing to divert from standard terminology. 
9973 is an arbitrary chosen prime number, the rationale for which is common knowledge.

File: src/nodemanagerbase.cpp (hunk 2 above)
Comment (pnikiel) [reply src/nodemanagerbase.cpp]: OK, Will do!

File: src/nodemanagerbase.cpp (hunk 3 above)
Comment (pnikiel) [reply src/nodemanagerbase.cpp:43]: I think this was really some unnecessary log output from the early days, I don't think it should stay.

**Pull request merged**